naming/src/main/java/com/alibaba/nacos/naming/misc/NamingExecutor.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.naming.misc;
+
+import com.alibaba.nacos.common.executor.ExecutorFactory;
+import com.alibaba.nacos.common.executor.NameThreadFactory;
+import com.alibaba.nacos.core.utils.ClassUtils;
+import com.alibaba.nacos.naming.NamingApp;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Global executor for naming
+ *
+ * <p>The classes provide the calling method of the thread pool, shielding the visibility of the thread pool itself, so
+ * that the outside can not operate its life cycle.
+ *
+ * @author gagharv
+ */
+@SuppressWarnings({"checkstyle:indentation", "PMD.ThreadPoolCreationRule"})
+public class NamingExecutor {
```

Comment:
Would chain calls be better?

for example 
```java
NamingExecutor.mysqlCheckExecutor().xxx
```

Reply:
This will cause the thread pool to be exposed to the outside but the user can call shutdown by himself.

Reply:
maybe this idea should implemented wrapped for ThreadPoolExecutor